src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -759,15 +784,21 @@ private int decodeProtocolHeader(
         final int protocolId = protocolHeader.id();
         int progress;
 
-        if (protocolId == 0)
+        switch (protocolId)
         {
+        case DEFAULT_PROTOCOL_ID:
```

Comment:
We might also consider calling this `PLAIN_PROTOCOL_ID` and renaming `decodeFrame` to `decodePlainFrame` as a peer of `decodeSaslFrame`.